src/Entities/Entity.cpp

```diff
@@ -1187,12 +1187,9 @@ void cEntity::TickBurning(cChunk & a_Chunk)
 	}
 
 	// Fire is extinguished by rain
-	if (GetWorld()->IsWeatherWetAt(POSX_TOINT, POSZ_TOINT))
+	if (GetWorld()->IsWeatherWetAtBlock({POSX_TOINT, POSY_TOINT, POSZ_TOINT}))
```

Comment:
If the entity falls out of the world or is above the world, this will pass invalid Y coord to the function. Although the above-world doesn't really matter, it still breaks the `IsWeatherWetAtBlock`'s contract about valid coords. The below-world will trigger an assert.